ribs/emitters/_gradient_aborescence_emitter.py

```diff
@@ -205,17 +205,29 @@ def ask(self):
             (batch_size, :attr:`solution_dim`) array -- a batch of new solutions
             to evaluate.
         """
-        lower_bounds = np.full(self._num_coefficients,
-                               -np.inf,
-                               dtype=self._archive.dtype)
-        upper_bounds = np.full(self._num_coefficients,
-                               np.inf,
-                               dtype=self._archive.dtype)
-        self._grad_coefficients = self.opt.ask(lower_bounds, upper_bounds)
-        noise = np.expand_dims(self._grad_coefficients, axis=2)
-
-        return self._grad_opt.theta + np.sum(
-            np.multiply(self._jacobian_batch, noise), axis=1)
+        coefficient_lower_bounds = np.full(self._num_coefficients,
+                                           -np.inf,
+                                           dtype=self._archive.dtype)
+        coefficient_upper_bounds = np.full(self._num_coefficients,
+                                           np.inf,
+                                           dtype=self._archive.dtype)
+
+        lower_bounds = np.expand_dims(self._lower_bounds, axis=0)
+        upper_bounds = np.expand_dims(self._upper_bounds, axis=0)
+
+        solution = None
+        # Sample gradient coefficient from CMA-ES until solution is within
+        # bounds.
+        while (solution is None or np.any(
```

Comment:
It's rather inefficient to keep resampling all the solutions -- in the CMA-ES optimizer, I resample only the solutions that are out of bounds -- take a look here: https://github.com/icaros-usc/pyribs/blob/master/ribs/emitters/opt/_cma_es.py#L206